lib/galaxy/webapps/galaxy/api/job_tokens.py

```diff
@@ -0,0 +1,78 @@
+""" API for asynchronous job running mechanisms can use to fetch or put files
+related to running and queued jobs.
+"""
+import logging
+
+from galaxy import (
+    exceptions,
+    model,
+    util,
+)
+from galaxy.authnz.util import provider_name_to_backend
+from galaxy.web import expose_api_raw_anonymous_and_sessionless
+from . import BaseGalaxyAPIController
+
+log = logging.getLogger(__name__)
+
+
+class JobTokensAPIController(BaseGalaxyAPIController):
+    """This job token controller allows remote job running mechanisms to
+    get a fresh OIDC token that can be used on remote side to authorize user.
+    It is not meant to represent part of Galaxy's stable, user
+    facing API.
+    """
+
+    @expose_api_raw_anonymous_and_sessionless
+    def get_token(self, trans, job_id, **kwargs):
+        """
+        GET /api/jobs/{job_id}/oidc-tokens
```

Comment:
Since you're just getting a single token `/api/jobs/{job_id}/oidc-token` might be preferable ?

Reply:
My thoughts were that we have multiple tokens and get only one due to the query parameter (`provider`).
So we basically call  `/api/jobs/{job_id}/oidc-tokens` with a filter (we do not return all tokens if no filter is provided, though).

But I can change it to `oidc-token` if this is a general convention in Galaxy, no problem.